### ISSUE TITLE
API Changes

### ISSUE THREAD

**Issue by KlausSchaefers** [eclipse/deeplearning4j#32]
Hi,

the API lacks a little bit of clarity for newcomers. Nothing serious, but it would be great if it would be more similar to standard ML tools, like scikit learn. May I suggest the following changes:
- DataSet API extends Pair<DoubleMatrix,DoubleMatrix>. To access the input I have to call getFirst(), for the label getSecond(). This should be renamed to getInstance() and getLabel() or so.
- One could think splitting label and instance data to x and y like sklearn. However this makes sorting / shuffling a little but more cumbersome. As other tools (weka, rapid miner, jstats) all have a similar API, its not such a big show stopper. 
- Change train() methods to fit() like in sklearn.
- There are quite a lot of classes for loaded the datasets. There are DataSet Iterators, Datasheets and some Vectorizers, which are somehow wrapped in the fetchers or so. 
  - I think there is no need for an extra DataFetcher object, because the BaseDataSetIterator just wraps it and adds batching. To keep the API simple it might be easier to just provide an AbstractDataSetIterator class with some abstract template method to user have to implement. Off course in the end the user just has to implement the iterator and he can compose it as he wants, but at least the default way should be straight forward. 
  - In general I think it is a good idea to encapsulate the vectorization process in an own object. However,
  - Why has the Fetcher and fetch(int count) and an next() method. Does it make sense to merge them to next(int size)? Or is this need for some parallel tasks?

Cheers,

Klaus
